### PR TITLE
Kotlin Conversion: PrefixSearchResponse

### PR DESCRIPTION
**Pull request by sharvaniharan**

### CODE REVIEW

File: app/src/main/java/org/wikipedia/search/PrefixSearchResponse.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package org.wikipedia.search
+
+import com.google.gson.annotations.SerializedName
+import org.wikipedia.dataclient.mwapi.MwQueryResponse
+
+class PrefixSearchResponse : MwQueryResponse() {
+    @SerializedName("searchinfo")
+    private val searchInfo: SearchInfo? = null
+    private val search: Search? = null
```
Comment (sharvaniharan) [inline app/src/main/java/org/wikipedia/search/PrefixSearchResponse.kt:9]: I feel we should leave the unused response attributes just for completion.. Lmk

File: app/src/main/java/org/wikipedia/search/PrefixSearchResponse.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    fun suggestion(): String? {
+        return searchInfo?.suggestion()
+    }
+
+    internal class SearchInfo {
+        private val suggestion: String? = null
+
+        @SerializedName("suggestionsnippet")
+        private val snippet: String? = null
+        fun suggestion(): String? {
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/search/PrefixSearchResponse.kt:20]: This does not need a getter function. We can simply make `suggestion` public.

File: app/src/main/java/org/wikipedia/search/PrefixSearchResponse.kt (hunk 1 above)
Comment (dbrant) [reply app/src/main/java/org/wikipedia/search/PrefixSearchResponse.kt:9]: agree

**Review by dbrant — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/wikipedia/search/PrefixSearchResponse.kt [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+    fun suggestion(): String? {
+        return searchInfo?.suggestion
+    }
+
+    internal class SearchInfo {
+        val suggestion: String? = null
+
+        @SerializedName("suggestionsnippet")
+        private val snippet: String? = null
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/search/PrefixSearchResponse.kt:19]: Would it be possible to shift `snippet` and to the top of `val suggestion: String? = null`?

**Pull request merged**